server/etcdserver/api/cluster.go

```diff
@@ -35,4 +35,6 @@ type Cluster interface {
 	Member(id types.ID) *membership.Member
 	// Version is the cluster-wide minimum major.minor version.
 	Version() *semver.Version
+
+	MustReadMembersFromBackend() (map[types.ID]*membership.Member, map[types.ID]bool)
```

Comment:
`Cluster` is a client-facing interface, It's weird to expose the backend functionality to the cluster interface directly. If it's only used by the test, then you can update the test something like below,

```
	be := schema.NewMembershipBackend(m.Logger, m.Server.Backend())
	members, removedMember := be.MustReadMembersFromBackend()

```

Reply:
good idea!